docker/docker-compose.yml

```diff
@@ -120,10 +131,18 @@ services:
       # todo : we are in process of making them as part of default.
       - PINOT_BROKER_TAG=DefaultTenant
       - PINOT_SERVER_TAG=DefaultTenant
-    restart: on-failure
+    volumes:
+      - ../scripts/wait-for-it.sh:/usr/bin/wait-for-it.sh
+    entrypoint: ["sh", "/usr/bin/wait-for-it.sh", "-h", "hypertrace-federated-service", "-p", "9002", "--timeout=120",
+                 "--" ,
+                 "java", "-cp", "/app/resources:/app/classes:/app/libs/*",
```

Comment:
you'll notice we don't have start scripts so what's happening here is we have to duplicate code, and this makes things extremely complex.

In the gradle plugin I think we should be adding `/usr/local/bin/start-platform` and using it as the entrypoint. In doing so, this simplifies immensely to `-- start-platform`. I'm already doing this with `start-kafka` in https://github.com/hypertrace/kafka/pull/8

another option for the change to the gradle config is to use the service name when generating the script. Ex.
`/usr/local/bin/start-${SERVICE_NAME}` which would be more intuitive in `docker ps`

cc @aaron-steinfeld as I've not raised a general PR on this, but have seen in a few cases people copy/pasting what gradle is generating, which is also lacking the ability to add `$JAVA_OPTS`.. if we add this feature to the gradle plugin, the entrypoint becomes more intuitive in ps and can be reused when people monkey with things like this.

Reply:
Moved java commands to its own script for now, and mounting it.

Reply:
java_opts are configurable like:
```
docker {
  javaApplication {
    jvmArgs.add("-Xmx=2gs")
  }
}
```
We should probably encapsulate it inside the ht plugin though. The broader point, about having an easy start script is fair.

Reply:
Thanks for mentioning how to change the default entrypoint. Ack the end goal is to be able to use JAVA_OPTS without overriding it (ex gradle writes a start-xxx script or even inlines entrypoint as `java ... $JAVA_OPTS ...`)